### PR TITLE
Add the app functionlaities to the view

### PR DESCRIPTION
**Pull request by Shaher-11**
In this milestone, we have set up the helper methods, controller, routes, and brought the functionalities to the browser as we have added the required buttons to the view.

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Shaher Mahmoud & Niko Escobal** 
Good job with your first submission for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] Your friendship model should have a method that creates a second row with inverted values of user_id and friend_id (and status equals true) when a friendship is confirmed. That way, you can find user friends only by user_id in friendships table.
- [x] Some of your tests are failing



- [x] root page should only shows post from the logged-in user and its friends.
- [x] The next erros appears in users route:



What I did was, create two users 'a' and 'b', 'b' request a friendship to user 'a', then login with user 'a' and click on 'ALL USERS' tab.

- [x] Please use the latest version of the linter.yml file for Ruby on Rails, remember not to edit it: https://github.com/microverseinc/linters-config/blob/master/ror/.github/workflows/linters.yml
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Comment by Shaher-11** [comment]
Done the requested all the requested changes as well as updating the linters file,
Thank you for the great feedback

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,9 +1,32 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
+  <%= invitation_button(@user) %>
+
+  <% if @user.friends.include?(current_user) %>
+  <h1>Friend</h1>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb]: Move this logic to a helper [ OPTIONAL ]

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+
+  <% if @user.inverse_friendships.find { |frd| frd.user_id == current_user.id && !frd.confirmed } %>
+  <h1>Pending Requests</h1>
+  <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb]: Move this logic to a helper [ OPTIONAL ]

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, Thank you for working on the changes suggested previously :ok_hand:
## To Highlight 🎉
- [x] You have mastered Git flow.
- [x] PR is descriptive and professional.
## To Improve 🙏

- [x] User should only see his posts and posts by his friends. The timeline is showing all the posts now. Please fix that!
- [x] You should add integration tests for friendship functionality. Kindly see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png). 
# Optional
- [x] App logic should be separate from views. You have some logic in the user show view template. You can move it to a helper.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by Shaher-11** [comment]
Enable user to see his own and friends posts as well as adding more integration tests and move the logic from view to application helper.

**Review by MkrtichSargsyan — APPROVED** [review]
## _Status:_ Approved  :hamster: :rocket:
Hello Shaher Mahmoud and Niko Escobal

Awesome job on this milestone.:rocket: Keep it up.
You have implemented all the changes that the previous Code Reviewers give you :clap:, 

You can go ahead and merge this branch to the base branch of this repo.

Good luck in your future projects and happy coding :relaxed:

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

**Pull request merged**